internal/command/clusters/uninstall.go

```diff
@@ -0,0 +1,349 @@
+		notifications = append(notifications, certManagerIntegrationsNotification)
+	}
+
+	return notifications, nil
+}
+
+// Notification holds information about a particular type of issue related to
+// uninstallation safety affecting a subset of resources in cluster
+type notification struct {
+	// header holds generic info about the issue and the suggested fix
+	header string
+	// listing of the affected resources with any additional related info
+	resourceInfos []string
+}
+
+func buildClients(kubeconfig *rest.Config) (allClients, error) {
+	secretsClient, err := clients.NewGenericClient[*corev1.Secret, *corev1.SecretList](
+		&clients.GenericClientOptions{
+			RestConfig: kubeconfig,
+			APIPath:    "/api/",
+			Group:      corev1.GroupName,
+			Version:    corev1.SchemeGroupVersion.Version,
+			Kind:       "secrets",
+		},
+	)
+	if err != nil {
+		return allClients{}, fmt.Errorf("error creating new secrets client: %w", err)
+	}
+
+	certsClient, err := clients.NewCertificateClient(kubeconfig)
+	if err != nil {
+		return allClients{}, fmt.Errorf("error creating new cert-manager.io client: %w", err)
+	}
+	podClient, err := clients.NewGenericClient[*corev1.Pod, *corev1.PodList](
+		&clients.GenericClientOptions{
+			RestConfig: kubeconfig,
+			APIPath:    "/api/",
+			Group:      corev1.GroupName,
+			Version:    corev1.SchemeGroupVersion.Version,
+			Kind:       "pods",
+		},
+	)
+	if err != nil {
+		return allClients{}, fmt.Errorf("error creating new pods client: %w", err)
+	}
+	return allClients{
+		certificates: certsClient,
+		secrets:      secretsClient,
+		pods:         podClient,
+	}, nil
+}
+
+type allClients struct {
+	secrets      clients.Generic[*corev1.Secret, *corev1.SecretList]
+	certificates clients.Generic[*cmapi.Certificate, *cmapi.CertificateList]
+	pods         clients.Generic[*corev1.Pod, *corev1.PodList]
+}
+
+func isUnready(cert cmapi.Certificate) bool {
+	hasReady := false
+	for _, cond := range cert.Status.Conditions {
+		if cond.Type == cmapi.CertificateConditionReady {
+			hasReady = true
+			if cond.Status == cmmeta.ConditionFalse {
+				return true
+			}
+		}
+	}
+	return !hasReady
+}
+
+func willBeRenewedSoon(cert cmapi.Certificate, buffer time.Duration, nowTime time.Time) bool {
+	return cert.Status.RenewalTime != nil && nowTime.Add(buffer).After(cert.Status.RenewalTime.Time)
+}
+
+func willExpireSoon(cert cmapi.Certificate, buffer time.Duration, nowTime time.Time) bool {
+	return cert.Status.NotAfter != nil && nowTime.Add(buffer).After(cert.Status.NotAfter.Time)
+}
+
+func isCurrentlyBeingIssued(cert cmapi.Certificate) bool {
+	for _, cond := range cert.Status.Conditions {
+		return cond.Type == cmapi.CertificateConditionIssuing && cond.Status == cmmeta.ConditionTrue
+	}
+	return false
+}
+
+func isCurrentlyFailingIssuance(cert cmapi.Certificate) bool {
+	failedAttempts := cert.Status.FailedIssuanceAttempts
+	return cert.Status.FailedIssuanceAttempts != nil && *failedAttempts > 0
```

Comment:
```suggestion
	return failedAttempts != nil && *failedAttempts > 0
```

Reply:
& also return this value, so it can be used by the parent function

Reply:
I prefer to have all the checker functions to work in the same way and return just a boolean. It makes it more readable. I don't see a problem with reading the failed attempts twice here

Reply:
I find it confusing that we first assign the value to a variable and then not use that variable.

Reply:
fixed in https://github.com/jetstack/jsctl/pull/71/commits/6478dbd89f3b4e8a18eca5421e6f76dfef42e3e8